packages/babel-make-styles/src/plugin.ts

```diff
@@ -0,0 +1,573 @@
+import { NodePath, PluginObj, PluginPass, types as t } from '@babel/core';
+import { declare } from '@babel/helper-plugin-utils';
+import { Module } from '@linaria/babel';
+import { MakeStyles, ResolvedStylesBySlots, resolveStyleRules } from '@fluentui/make-styles';
+
+import { astify } from './utils/astify';
+import { evaluatePaths } from './utils/evaluatePaths';
+
+type AstStyleNode =
+  | { kind: 'PURE_OBJECT'; nodePath: NodePath<t.ObjectExpression> }
+  | {
+      kind: 'LAZY_OBJECT';
+      nodePath: NodePath<t.ObjectExpression>;
+      lazyPaths: NodePath<t.Expression | t.SpreadElement>[];
+    }
+  | { kind: 'LAZY_FUNCTION'; nodePath: NodePath<t.ArrowFunctionExpression | t.FunctionExpression> }
+  | { kind: 'LAZY_IDENTIFIER'; nodePath: NodePath<t.Identifier> }
+  | { kind: 'SPREAD'; nodePath: NodePath<t.SpreadElement>; spreadPath: NodePath<t.SpreadElement> };
+
+type BabelPluginState = PluginPass & {
+  importDeclarationPath?: NodePath<t.ImportDeclaration>;
+  requireDeclarationPath?: NodePath<t.VariableDeclarator>;
+
+  /** Contains all paths to calls of makeStyles(). */
+  calleePaths?: NodePath<t.Identifier>[];
+
+  /** Contains AST nodes with that should be resolved. */
+  styleNodes?: AstStyleNode[];
+};
+
+/**
+ * Parses a MemberExpression that defines token usage for extract parts that are needed to generate a CSS variable.
+ *
+ * @example
+ *   getTokenParts(theme.alias.color.green.foreground1)
+ *   // results in
+ *   ['alias', 'color', 'green', 'foreground']
+ */
+function getTokenParts(path: NodePath<t.MemberExpression>, result: string[] = []): string[] {
+  const objectPath = path.get('object');
+  const propertyPath = path.get('property');
+
+  if (objectPath.isMemberExpression()) {
+    getTokenParts(objectPath, result);
+  } else if (objectPath.isIdentifier()) {
+    // We ignore this part as it will return a first id (i.e. "theme")
+  } else {
+    throw new Error('We met an unhandled case, this is a bug, please report it');
```

Comment:
nit: This happens quite a few times, let's create a custom error for it

```suggestion
    throw new UnhandledCaseError();
```

Reply:
In a case when we have `*Path`, we should not throw generic `Error` at all, fixed in 1352c27